### DISCUSSION TITLE
Transpiling React Flow

### DISCUSSION THREAD

**Discussion by sweinstein22** [xyflow/xyflow#2795] [Q&A] [answered]
Hi Everyone,

I'm looking to use React Flow in a project that isn't bootstrapped off of Create React App or any other framework, which has made adding React Flow challenging as I need to know how to compile the package.

It would be great if I could get a list of the transpilers React Flow expects to be at play. It looks like just having babel-loader isn't sufficient

Errors include the following:
```
Unexpected token (316:78)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
You may need an additional loader to handle the result of these loaders.
   const evtY = isMouseTriggered ? event.clientY : (_event$touches2 = event.touches) === null || _event$touches2 === void 0 ? void 0 : _event$touches2[0].clientY;
   return {
>     x: evtX - ((bounds === null || bounds === void 0 ? void 0 : bounds.left) ?? 0),
      y: evtY - ((bounds === null || bounds === void 0 ? void 0 : bounds.top) ?? 0)
   };
```

Additionally I have two questions for the maintainers:
1) Is there a reason y'all stopped shipping a compiled version of React Flow? It would be super helpful to have a version of things that makes setup easier when leveraging the package!
2) Would it be possible to ship `.scss` files as well as the `.css` files for projects using sass?

Thank you for your time!

**Comment by moklick** [comment, marked as answer]
Hey @sweinstein22 

1. babel
you can use babel to transpile React Flow but you need to make sure that @reactflow/.. and reactflow packages are included in the transpilation process. Maybe this helps https://github.com/wbkd/react-flow/issues/2453#issuecomment-1262046993

2. scss
we are not using scss and so we are not providing scss files

**Reply by sweinstein22** [reply]
Ah that helped! Instead of explicitly excluding all node modules except react flow I added an explicit rule so that we always use `babel-loader` for reactflow. Hopefully that'll also help future proof things in case we make other adjustments in the future. Thanks for your time!

For anyone else, the rule I added was:
``` 
module.exports = config => {
  config.module = {
    rules: [
      ...
      {
        test: /\.js$/,
        include: /node_modules[\\/]@?reactflow/,
        use: {
          loader: "babel-loader",
        },
      },
    ],
  };
};
```

**Reply by MrVibe** [reply]
Thank you this was helpful. But I now see another error.


`Uncaught TypeError: Cannot read properties of undefined (reading '__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED')
    at eval (react-jsx-runtime.development.js:52:34)
    at eval (react-jsx-runtime.development.js:1313:5)
    at ./node_modules/react/cjs/react-jsx-runtime.development.js (workflows.js?ver=1418:1908:1)
    at __webpack_require__ (workflows.js?ver=1418:20:30)
    at eval (jsx-runtime.js:4:20)
    at ./node_modules/react/jsx-runtime.js (workflows.js?ver=1418:1920:1)
    at __webpack_require__ (workflows.js?ver=1418:20:30)
    at eval (index.js:56:75)`

 The only difference in your webpack and mine is that I use external react as well in my webpack. 

`externals: {
    'react-dom': {
      root: 'ReactDOM',
      commonjs2: 'react-dom',
      commonjs: 'react-dom',
      amd: 'react-dom',
      umd: 'react-dom',
    },
    react: {
      root: 'React',
      commonjs2: 'react',
      commonjs: 'react',
      amd: 'react',
      umd: 'react',
    }
  },`

any help would be deeply appreciated.

**Reply by moklick** [reply]
which React version are you using?

**Reply by MrVibe** [reply]
I am using 18.2 , I need to power it via external react not bundled in the react package.
